hmcts/pre-api
# S28 3225: Search Bookings By Capture Session Status

### JIRA ticket(s)
- S28-3225


### Change description
- Add search param `captureSessionStatusIn` to `GET /bookings`
- Add search param `captureSessionStatusNotIn` to `GET /bookings`




> [!IMPORTANT]
> This PR requires manual testing by QA. Please notify the QA team @hmcts/pre-rec-evidence-qa.

**QA instructions**
- Create a booking each with one capture session in a different status (`INITIALISING`, `STANDBY`, `RECORDING`, `PROCESSING`, `FAILURE`, `NO_RECORDING`, `RECORDING_AVAILABLE`) and one booking without an associated capture session (total: 8 bookings, 7 capture sessions)
- Test filtering by capture session status in (`captureSessionStatusIn`)- should return only the bookings with capture sessions of the specified statuses (i.e. filtering by `STANDBY` and `RECORDING` will return the bookings that are associated with capture sessions in either the `STANDBY` or `RECORDING` status)
- Test filtering by capture session status not in (`captureSessionStatusNotIn`)- should return only the bookings with capture sessions of the specified statuses OR bookings without a capture session associated(i.e. filtering by `STANDBY` and `RECORDING` will return the bookings that are associated with capture sessions NOT IN either the `STANDBY` or `RECORDING` status and will include any bookings without associated capture sessions).
- Emulate real world use of these params:
    - New bookings screen: Search bookings by `?captureSessionStatusNotIn=PROCESSING,FAILURE,NO_RECORDING,RECORDING_AVAILABLE`
    - New processing/failure screen: Search bookings by `?captureSessionStatusIn=FAILURE,PROCESSING`
